ubcpi/answer_pool.py

```diff
@@ -0,0 +1,315 @@
+import random
+import persistence as sas_api
+
+POOL_SIZE = 50
+
+
+class UnknownChooseAnswerAlgorithm(Exception):
+    pass
+
+
+def offer_answer(pool, answer, rationale, student_id, algo):
+    """
+    submit a student answer to the answer pool
+
+    The answer maybe selected to stay in the pool depending on the selection algorithm
+
+    Args:
+        pool (dict): answer pool
+            Answer pool format:
+            {
+                option1_index: {
+                    'student_id': { can store algorithm specific info here },
+                    ...
+                }
+                option2_index: ...
+            }
```

Comment:
Hmm, sorry I totally misunderstood that; my notes say "answer pool size is 50 responses", nothing about per option. Sorry I missed your earlier comment.

I would prefer to see an enforcement on the total size of the pool, rather than per entry, because that can grow without bound in extreme cases. If the total size of the pool, across _all_ problems, is 50 entries, your pool is about 785 bytes. If it's 100, we're up at 1.3k or so. Remember that on every block render, the block will load the data, so that's loading 1.3k per every page view. We haven't stressed XBlocks in this way and since time is so short, it's a bit worrying. We don't really have time for extensive load testing.

As I see it we have two options: limit the total numbers of entries in the pool, or limit the overall size of the pool (to maybe 2k?). The second option is probably the better option, because as you add more data to each entry you don't have to worry about how additional fields are stressing the size of this field.